src/cpp/session/modules/SessionHelp.R

```diff
@@ -738,26 +739,30 @@ options(help_type = "html")
                                     subset = TRUE,
                                     getSignature = FALSE)
 {
-   # Ensure topic and package are not zero-length
+   # ensure topic and package are not zero-length
    if (!length(package))
       package <- ""
    
    if (!length(topic))
       topic <- ""
    
-   # Completions from the search path might have the 'package:' prefix, so
-   # lets strip that out.
+   # completions from the search path might have the 'package:' prefix;
+   # let's strip that out
    package <- sub("package:", "", package, fixed = TRUE)
    
-   # If the package is not provided, but we're getting help on e.g.
-   # 'stats::rnorm', then split up the topic into the appropriate pieces.
+   # if the package is not provided, but we're getting help on e.g.
+   # 'stats::rnorm', then split up the topic into the appropriate pieces
    if (package == "" && any(grepl(":{2,3}", topic, perl = TRUE)))
    {
       splat <- strsplit(topic, ":{2,3}", perl = TRUE)[[1]]
       topic <- splat[[2]]
       package <- splat[[1]]
    }
    
+   # don't provide help for objects in the global environment
+   if (identical(package, ".GlobalEnv"))
+      return()
+   
```

Comment:
I'm sort of on the fence about this one. If you do something like:

```
data(iris)
iri<TAB>
```

You won't see any help for the `iris` dataset pop up, as RStudio isn't able to figure out this `iris` object in the global environment is just a copy of the `iris` dataset that lives in the `datasets` package. This might be worth tweaking / fixing.